akka-stream-testkit/src/main/scala/akka/stream/testkit/StreamTestKit.scala

```diff
@@ -309,6 +317,9 @@ object TestSubscriber {
       self
     }
 
+    def expectMsgPF[T](f: PartialFunction[AnyRef, T]): T =
+      f(probe.expectMsgType[AnyRef])
```

Comment:
first, you have to check f.isDefinedAt(_).

Check out the actor version here: https://github.com/akka/akka/blob/master/akka-testkit/src/main/scala/akka/testkit/TestKit.scala#L365